Material.Blazor/Components/Button/MBButton.razor

```diff
@@ -2,14 +2,48 @@
 @inherits ComponentFoundation
 
 
-<div class="mdc-touch-target-wrapper">
+@if (AppliedTouchTarget)
+{
+    <div class="mdc-touch-target-wrapper">
+        <button @ref="@ElementReference"
+                class="mdc-button @ActiveConditionalClasses @(@class)"
+                style="@style"
+                id="@id"
+                @attributes="@AttributesToSplat()"
+                data-mdc-dialog-action="@DialogAction">
+
+            <div class="mdc-button__ripple" />
+
+            @if (!string.IsNullOrWhiteSpace(LeadingIcon))
+            {
+                <MBIcon aria-hidden="true"
+                        class="mdc-button__icon"
+                        IconFoundry="@IconFoundry"
+                        IconName="@LeadingIcon" />
+            }
+
+            <span class="mdc-button__label">@Label</span>
+
+            @if (!string.IsNullOrWhiteSpace(TrailingIcon))
+            {
+                <MBIcon aria-hidden="true"
+                        class="mdc-button__icon"
+                        IconFoundry="@IconFoundry"
+                        IconName="@TrailingIcon" />
+            }
+        </button>
```

Comment:
The inner markup is identical to the `else` branch, isn't it?

Reply:
This approach prevents any duplication of code (which we should avoid, because it's harder to maintain)

Reply:
I actually did that first (You also have to catch the closing div) but I thought it was obscuring the intent. It's not like we change the markup frequently.

Reply:
You don't want "@<text>" just "<text>"

Also, this code doesn't work as written; Note the two buttons in the following image:

![chrome_2021-03-29_12-04-00](https://user-images.githubusercontent.com/634561/112865670-faa3fa80-9086-11eb-805b-1abec30c7327.png)

Reply:
The code above works perfectly fine, even though I expected it to have a typo because I wrote it on Github without checking that it compiles: https://blazorrepl.com/repl/GvudctFA11vZJZcy19

Reply:
I pasted the code back into the button component & it failed in the same fashion, requiring me to remove the "@" before the text component. I then got suspicious, exited VS, restarted, but the "@" symbol back and everrything works. So I was observing yet another VS failure in Razor code compiling.

Reply:
In Checkbox, I ran into an issue where the text of the label was getting wrapped when lengthy and in a div with 'display: flex'. It turns out that is 'job' of 'mdc-form-field'. The following is it's css which I am finding somewhat impenetrable, I was hoping that a simple override might have taken case of the issue:
```
.mdc-form-field {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-family: Roboto, sans-serif;
  /* @alternate */
  font-family: var(--mdc-typography-body2-font-family, var(--mdc-typography-font-family, Roboto, sans-serif));
  font-size: 0.875rem;
  /* @alternate */
  font-size: var(--mdc-typography-body2-font-size, 0.875rem);
  line-height: 1.25rem;
  /* @alternate */
  line-height: var(--mdc-typography-body2-line-height, 1.25rem);
  font-weight: 400;
  /* @alternate */
  font-weight: var(--mdc-typography-body2-font-weight, 400);
  letter-spacing: 0.0178571429em;
  /* @alternate */
  letter-spacing: var(--mdc-typography-body2-letter-spacing, 0.0178571429em);
  text-decoration: inherit;
  /* @alternate */
  -webkit-text-decoration: var(--mdc-typography-body2-text-decoration, inherit);
  text-decoration: var(--mdc-typography-body2-text-decoration, inherit);
  text-transform: inherit;
  /* @alternate */
  text-transform: var(--mdc-typography-body2-text-transform, inherit);
  color: rgba(0, 0, 0, 0.87);
  /* @alternate */
  color: var(--mdc-theme-text-primary-on-background, rgba(0, 0, 0, 0.87));
  display: inline-flex;
  align-items: center;
  vertical-align: middle;
}

.mdc-form-field > label {
  /* @noflip */
  margin-left: 0;
  /* @noflip */
  margin-right: auto;
  /* @noflip */
  padding-left: 4px;
  /* @noflip */
  padding-right: 0;
  order: 0;
}

[dir=rtl] .mdc-form-field > label, .mdc-form-field > label[dir=rtl] {
  /* @noflip */
  margin-left: auto;
  /* @noflip */
  margin-right: 0;
}

[dir=rtl] .mdc-form-field > label, .mdc-form-field > label[dir=rtl] {
  /* @noflip */
  padding-left: 0;
  /* @noflip */
  padding-right: 4px;
}

.mdc-form-field--nowrap > label {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.mdc-form-field--align-end > label {
  /* @noflip */
  margin-left: auto;
  /* @noflip */
  margin-right: 0;
  /* @noflip */
  padding-left: 0;
  /* @noflip */
  padding-right: 4px;
  order: -1;
}

[dir=rtl] .mdc-form-field--align-end > label, .mdc-form-field--align-end > label[dir=rtl] {
  /* @noflip */
  margin-left: 0;
  /* @noflip */
  margin-right: auto;
}

[dir=rtl] .mdc-form-field--align-end > label, .mdc-form-field--align-end > label[dir=rtl] {
  /* @noflip */
  padding-left: 4px;
  /* @noflip */
  padding-right: 0;
}

.mdc-form-field--space-between {
  justify-content: space-between;
}

.mdc-form-field--space-between > label {
  margin: 0;
}

[dir=rtl] .mdc-form-field--space-between > label, .mdc-form-field--space-between > label[dir=rtl] {
  margin: 0;
}
```